File: frontend/src/api/customer.ts
import request from './request';

// ── 类型 ──
export interface CustomerListItem {
  id: string;
  code: string;
  name: string;
  shortName?: string;
  contactPerson?: string;
  contactPhone?: string;
  isActive: boolean;
  createdAt: string;
}

export interface CustomerDetail extends CustomerListItem {
  remark?: string;
  updatedAt?: string;
}

export interface CustomerPagedResult {
  items: CustomerListItem[];
  total: number;
  page: number;
  pageSize: number;
}

export interface CustomerQuery {
  keyword?: string;
  code?: string;
  isActive?: boolean;
  page: number;
  pageSize: number;
}

export interface CustomerFormData {
  name: string;
  shortName?: string;
  contactPerson?: string;
  contactPhone?: string;
  remark?: string;
  isActive: boolean;
}

// ── API ──
// 注：request.ts 响应拦截器返回 response.data，故使用双泛型 <unknown, T>
// 使 Promise 解析为 T（与 numbering.ts / role.ts / user.ts 一致）。
export function getCustomers(params: CustomerQuery) {
  return request.get<unknown, CustomerPagedResult>('/api/customers', { params });
}

export function getCustomer(id: string) {
  return request.get<unknown, CustomerDetail>(`/api/customers/${id}`);
}

export function createCustomer(data: CustomerFormData) {
  return request.post<unknown, CustomerDetail>('/api/customers', data);
}

export function updateCustomer(id: string, data: CustomerFormData) {
  return request.put<unknown, CustomerDetail>(`/api/customers/${id}`, data);
}

export function deleteCustomer(id: string) {
  return request.delete(`/api/customers/${id}`);
}

File: backend/src/OneCup.Domain/Entities/Customer.cs
namespace OneCup.Domain.Entities;

/// <summary>
/// 客户档案。客户编号由编号系统在创建事务内生成。
/// </summary>
public class Customer : BaseEntity, ISoftDeletable
{
    /// <summary>客户编号（编号系统生成，如 CUST-0001）</summary>
    public string Code { get; set; } = string.Empty;

    /// <summary>客户名称（全名，唯一）</summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>客户简称（可重复）</summary>
    public string? ShortName { get; set; }

    /// <summary>联系人</summary>
    public string? ContactPerson { get; set; }

    /// <summary>联系电话</summary>
    public string? ContactPhone { get; set; }

    /// <summary>备注</summary>
    public string? Remark { get; set; }

    /// <summary>启用状态（停用的客户不再用于新业务，但保留历史）</summary>
    public bool IsActive { get; set; } = true;

    public bool IsDeleted { get; set; } = false;
}
